docs/proposals/20220927-label-sync-between-machine-and-nodes.md

```diff
@@ -0,0 +1,137 @@
+
+### Implementation Details/Notes/Constraints
+
+While designing a solution for syncing labels between Machine and underlying Kubernetes Nodes two main concerns have been considered:
+
+- Security, because Node labels can be used to schedule and/or limit workloads to a predetermined set of nodes.
+- Impact on other components applying labels on Kubernetes Nodes, like e.g. Kubeadm, CPI etc.
+
+### Label Prefixes
+
+The idea of scoping synchronization to a well defined set of labels is a first answer to security/concurrency concerns; labels to be managed by Cluster API have been selected based on following criteria:
+
+- The `node-role.kubernetes.io` label has been used widely in the past to identify the role of a Kubernetes Node (e.g. `node-role.kubernetes.io/worker=''`). For example, `kubectl get node` looks for this specific label when displaying the role to the user.
+
+- The `node-restriction.kubernetes.io/` domain is recommended in the [Kubernetes docs](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction) for things such as workload placement; please note that in this case
+  we are considering the entire domain, thus also labels like `example.node-restriction.kubernetes.io/fips=true` falls in this category.
+
+- Cluster API owns a specific domain: `node.cluster.x-k8s.io`.
+
+#### Synchronization of CAPI Labels
+
+The synchronization of labels between a Machine object and its linked Node is limited to the domains and prefixes described in the section above.
+
+The synchronization process is going to use [server side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) in order to ensure that Cluster API will manage only the subset of labels coming from Machine objects and ignore labels applied concurrently by other users/other components.
+
+This requires to define an identity/manager name to be used by CAPI when performing this operation; additionally during implementation we are going to identify and address eventual additional steps required to properly transition existing Nodes to SSA, if required.
+From a preliminary investigation, the risk is that the manager that created the object and the new SSA manager will become co-owner of the same labels, and this will prevent deletion of labels.
+Similar risks applies to clusterctl move or operations like velero backup/restore. 
+
+The syncronization process will be implemented in the Machine controller. Reconciliation is triggered both when Machine object changes, or when the Node changes (the machine controller watches Nodes within the workload cluster).
+
+#### Delay between Node Create and Label Sync
+
+The Node object is first created when kubeadm joins a node to the workload cluster (i.e. kubelet is up and running). There may be a delay (potentially several seconds) before the machine controller kicks in to apply the labels on the Node.
```

Comment:
just a nit but I wouldn't specify kubeadm here as regardless the bootstrapping choice ultimately the kubelet is responsible for specifying the labels on node creation.